webapp/src/components/chat/tabs/PlansTab.tsx

```diff
@@ -1,13 +1,180 @@
+                        <TableRow key={item.ask}>{columns.map((column) => column.renderCell(item))}</TableRow>
+                    ))}
+                </TableBody>
+            </Table>
+        </TabView>
     );
 };
+
+function useTable(planMessages: IChatMessage[]) {
+    const headerSortProps = (columnId: TableColumnId): TableHeaderCellProps => ({
+        onClick: (e: React.MouseEvent) => {
+            toggleColumnSort(e, columnId);
+        },
+        sortDirection: getSortDirection(columnId),
+    });
+
+    const columns: Array<TableColumnDefinition<TableItem>> = [
+        createTableColumn<TableItem>({
+            columnId: 'ask',
+            renderHeaderCell: () => (
+                <TableHeaderCell key="ask" {...headerSortProps('ask')}>
+                    Ask
+                </TableHeaderCell>
+            ),
+            renderCell: (item) => (
+                <TableCell key={`plan-${item.index}`}>
+                    <TableCellLayout>
+                        <PlanJsonViewer goal={item.ask} json={item.message.content} />
+                    </TableCellLayout>
+                </TableCell>
+            ),
+            compare: (a, b) => {
+                const comparison = a.ask.localeCompare(b.ask);
+                return getSortDirection('name') === 'ascending' ? comparison : comparison * -1;
+            },
+        }),
+        createTableColumn<TableItem>({
+            columnId: 'createdOn',
+            renderHeaderCell: () => (
+                <TableHeaderCell key="createdOn" {...headerSortProps('createdOn')}>
+                    Created on
+                </TableHeaderCell>
+            ),
+            renderCell: (item) => (
+                <TableCell key={item.createdOn.timestamp} title={new Date(item.createdOn.timestamp).toLocaleString()}>
+                    {item.createdOn.label}
+                </TableCell>
+            ),
+            compare: (a, b) => {
+                const comparison = a.createdOn.timestamp > b.createdOn.timestamp ? 1 : -1;
+                return getSortDirection('createdOn') === 'ascending' ? comparison : comparison * -1;
+            },
+        }),
+        createTableColumn<TableItem>({
+            columnId: 'tokenCounts',
+            renderHeaderCell: () => (
+                <TableHeaderCell key="tokenCounts" {...headerSortProps('tokenCounts')}>
+                    Token Count
+                </TableHeaderCell>
+            ),
+            renderCell: (item) => <TableCell key={`plan-${item.index}-tokens`}>{item.tokens}</TableCell>,
+            compare: (a, b) => {
+                const comparison = a.tokens - b.tokens;
+                return getSortDirection('tokenCounts') === 'ascending' ? comparison : comparison * -1;
+            },
+        }),
+    ];
+
+    /*
+    eslint-disable 
+        @typescript-eslint/no-unsafe-assignment,
+        @typescript-eslint/no-unsafe-member-access,
+        @typescript-eslint/no-unsafe-call,
+    */
```

Comment:
can we add a comment as to why we need these?

Reply:
Added a TODO to reenable these checks.